spec/requests/companies_spec.rb

```diff
@@ -4,9 +4,19 @@
 
 RSpec.describe 'Companies', type: :request do
   describe 'GET /index' do
-    it 'returns http success' do
+    it 'render template' do
       get '/companies'
+
       expect(response).to have_http_status(:success)
+      expect(response).to render_template(:index)
+      expect(response.body).to include('Companies')
+      expect(response.body).to include('Joint Stock Company')
+    end
+
+    it 'does not render a different template' do
+      get '/companies'
+
+      expect(response).to_not render_template(:show)
     end
   end
 end
```

Comment:
The test for `/companies/:id` is missing. I think you should have at least one "requests" test for every endpoint.

Reply:
Done :)